.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+      - '**.md' #Do not need to run CI for markdown changes.
+  pull_request:
+    branches: [main]
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  build-test:
+    strategy:
+      matrix:
+        versions: 
+        - {os: 'ubuntu-20.04', otp: '21.x', rebar: '3.15.2'}
+        - {os: 'ubuntu-22.04', otp: '24.x', rebar: '3.18.0'}
+        - {os: 'ubuntu-22.04', otp: '25.x', rebar: '3.18.0'}
+            
+    runs-on: ${{ matrix.versions.os }}
+    name: Build and Test - ${{ matrix.versions.otp }}
+
+    steps:
+      - uses: erlef/setup-beam@v1
```

Comment:
I was using containers, but by not using a container it allows us to simplify how we configure redis, and more easily compose parts of the builds together.